app/models/post.rb

```diff
@@ -1,2 +1,14 @@
 class Post < ApplicationRecord
+  has_many :likes, foreign_key: post_id
+  has_many :comments, foreign_key: :post_id
+
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] Please call the method `update_post_counter` after save, It will update the `posts_counter` for the author when a post is created.
  - You can consider the below code.
``` ruby
after_save :update_post_counter